fairscale/nn/pipe/pipeline.py

```diff
@@ -180,7 +180,10 @@ def function(
             part_id: int = j,
         ) -> TensorOrTensors:
             with use_skip_tracker(skip_tracker), record_function("chunk%d-part%d" % (chunk_id, part_id)):
-                return partition(input)
+                ret = partition(input)
+                if type(ret) is list:
```

Comment:
I'm not sure about this check. It will be run for every partition for every micro-batch.

Reply:
Yeah, was worrying about that too. How about I turn it into a comment?